### DISCUSSION TITLE
Support NovoPen 6 / Echo Plus

### DISCUSSION THREAD

**Discussion by damiculus** [JohanDegraeve/xdripswift#591] [Ideas]
xDrip from NightscoutFoundation supports scanning smart insulin pens Novopen 6 / Echo Plus from Novo Nordisk. Is it possible to implement this in this project?

https://github.com/NightscoutFoundation/xDrip/issues/2074

I volunteer to help, I have experience as a developer, but I don't have a lot of experience with Swift, and I'm currently learning, if there is any coding task needed I can do something simple. I have an Echo Plus pen that I use daily for MDI diabetes management, that I can help testing with; an iPhone and a Mac with Xcode if needed; and an Apple Developer paid license.

**Comment by m-a-v** [comment]
> **Prime the insulin pen**. Priming means removing air bubbles from the needle. It ensures that the needle is open and working. You must prime the pen before each injection. To prime the insulin pen, turn the dosage knob to the 2 units indicator. With the pen pointing upward, push the knob all the way. At least one drop of insulin should appear. You may need to repeat this step until a drop appears.
Source: https://my.clevelandclinic.org/health/treatments/17923-insulin-pen-injections

Great idea! One challenge I foresee is that the insulin priming before transferring values to Nightscout needs to be excluded. This task seems quite complex from a UI/UX perspective, especially if multiple injections from the current and eventually past day are being transferred.

**Comment by gui-dos** [comment]
@damianocsl:

In LibreLink’s `Info.plist` you can spot the key

```
    <key>com.apple.developer.nfc.readersession.iso7816.select-identifiers</key>
    <array>
      <string>D2760000850101</string>
    </array>
```

so it is clear that smart pens adopt a different NFC protocol than Libre’s NFC ISO 15693 (https://developer.apple.com/documentation/corenfc/nfciso7816tag) which will require a substantial rewrite of the current NFC support.

**Comment by dgow** [comment]
Any progress here? Can't find any useful apps that can do this.
